packages/@lwc/template-compiler/src/parser/expression-complex/html.ts

```diff
@@ -226,14 +261,21 @@ class TemplateHtmlParser extends Parser {
     // that, we create a new text node for the template expression rather than
     // allowing the concatenation to proceed.
     _insertCharacters(token: any) {
-        if (token.chars[0] !== '{') {
+        const parentNode = this.openElements.current;
+        const previousPeer = parentNode.childNodes.at(-1);
```

Comment:
Confirm that `Array.prototype.at` can be used in all environments that `@lwc/template-compiler` will be running (Node 16+).